### PR TITLE
refactor(runtime): sealos runtime module needs better design to meet …

### PR DESCRIPTION
**Pull request by cubxxw**
…future k3s and k0s...

This PR is still working on #1943, #2037, please note

Please give suggestions in this PR and follow up this PR

This is part of the plan: #2863

### CODE REVIEW

File: pkg/guest/guest.go [hunk 1]
```diff
@@ -22,7 +22,7 @@ import (
 	"github.com/labring/sealos/fork/golang/expansion"
 	"github.com/labring/sealos/pkg/constants"
 	"github.com/labring/sealos/pkg/env"
-	"github.com/labring/sealos/pkg/runtime"
+	runtime "github.com/labring/sealos/pkg/runtime/kubernetes"
```
Comment (cuisongliu) [inline pkg/guest/guest.go:25]: Can you also change it here, the guest should not pay attention to the real-time details of the runtime

File: pkg/apply/processor/install.go [hunk 2]
```diff
@@ -166,7 +167,7 @@ func (c *InstallProcessor) PreProcess(cluster *v2.Cluster) error {
 		cluster.SetMountImage(mount)
 		c.NewMounts = append(c.NewMounts, *mount)
 	}
-	runtime, err := runtime.NewDefaultRuntime(cluster, c.ClusterFile.GetKubeadmConfig())
+	runtime, err := kubernetes.NewDefaultRuntime(cluster, c.ClusterFile.GetKubeadmConfig())
```
Comment (cuisongliu) [inline pkg/apply/processor/install.go:170]: NewDefaultRuntime should be in `github.com/labring/sealos/pkg/runtime` .

Comment (cubxxw) [reply pkg/apply/processor/install.go:170]: yep

File: cmd/sealctl/cmd/token.go [hunk 3]
```diff
@@ -30,7 +30,8 @@ func newTokenCmd() *cobra.Command {
 		Use:   "token",
 		Short: "token generator",
 		Run: func(cmd *cobra.Command, args []string) {
-			t, err := runtime.Generator()
+			//TODO: Whether a token is required to encapsulate an interface?
+			t, err := kubernetes.Generator()
```
Comment (fanux) [inline cmd/sealctl/cmd/token.go:34]: So the caller should know runtime is kubernetes?

File: cmd/sealctl/cmd/token.go [hunk 4]
```diff
@@ -30,7 +30,8 @@ func newTokenCmd() *cobra.Command {
...
 			if err != nil {
 				logger.Error("exec token error: " + err.Error())
 				os.Exit(1)
```
Comment (fanux) [inline cmd/sealctl/cmd/token.go:37]: Use RunE to instead Exit

File: pkg/apply/gen.go [hunk 5]
```diff
@@ -21,7 +21,7 @@ import (
 
 	"github.com/labring/sealos/pkg/apply/processor"
 	"github.com/labring/sealos/pkg/buildah"
-	"github.com/labring/sealos/pkg/runtime"
+	runtime "github.com/labring/sealos/pkg/runtime/kubernetes"
```
Comment (fanux) [inline pkg/apply/gen.go:24]: Maybe use NewRuntime(runtime type) is better than use kuberntes package directly.

File: pkg/apply/processor/create.go [hunk 6]
```diff
@@ -89,7 +90,7 @@ func (c *CreateProcessor) PreProcess(cluster *v2.Cluster) error {
 	if err := MountClusterImages(cluster, c.Buildah); err != nil {
 		return err
 	}
-	runTime, err := runtime.NewDefaultRuntime(cluster, c.ClusterFile.GetKubeadmConfig())
+	runTime, err := kubernetes.NewDefaultRuntime(cluster, c.ClusterFile.GetKubeadmConfig())
```
Comment (fanux) [inline pkg/apply/processor/create.go:93]: runtime.NewRuntime(runtimetype string,....)

**Comment by cuisongliu** [comment]
add `sealos.io.runtime="k8s"` to spec.

File: cmd/sealctl/cmd/token.go (hunk 3 above)
Comment (cubxxw) [reply cmd/sealctl/cmd/token.go:34]: I don't need to know. I'll just move the table of contents and adjust it later

File: pkg/apply/processor/create.go (hunk 6 above)
Comment (cubxxw) [reply pkg/apply/processor/create.go:93]: NewRuntime --> Should I put it in runtime or outside?

**Pull request closed without merging**